### ISSUE TITLE
Add a manual override for setting proposal expired status

### ISSUE THREAD

**Issue by Comeani** [pitt-crc/bank#309]
I was working on updates to the CRC bank SOP and noticed that if you create a proposal, and then create another proposal immediately afterwards, the only ways to have the proposal you created second considered the active proposal is to delete the first proposal or manually set it's allocation values/end date such that it would trigger `isActive == false`:

```
(bank_deploy) [nlc60@moss bank] db-new-account-insert : crc-bank proposal create test --all-clusters 25000

(bank_deploy) [nlc60@moss bank] db-new-account-insert : crc-bank proposal create test --smp 10000 --gpu 10000
```
In the old bank, it would be expected that the first proposal would be moved to the archive table, with the newest proposal set as the new one to actively draw resources from. This was useful in the case that a standard allocation was being replaced with a proposal before it's end date.  

The new bank picks up the standard allocation instead of replacing it: 
```
(bank_deploy) [nlc60@moss bank] db-new-account-insert : crc-bank account info test
+-------------------------------------------------------------------+-------------------------+----------------------+
|                         Proposal End Date:                        |        07/26/2024       |                      |
+-------------------------------------------------------------------+-------------------------+----------------------+
|                            Proposal ID:                           |            1            |                      |
+-------------------------------------------------------------------+-------------------------+----------------------+
|                                                                   |                         |                      |
+-------------------------------------------------------------------+-------------------------+----------------------+
|                       Floating Service Units                      |     Units Remaining     |     Percent Used     |
+-------------------------------------------------------------------+-------------------------+----------------------+
|     **Floating SUs are applied to any cluster to cover usage      |          25000          |          0           |
|                     exceeding proposal limits                     |                         |                      |
+-------------------------------------------------------------------+-------------------------+----------------------+
|                          Aggregate Usage                          |            0            |                      |
+-------------------------------------------------------------------+-------------------------+----------------------+
```

Adjusting the date so the standard allocation is expired allows the other proposal to take the active position:

```
(bank_deploy) [nlc60@moss bank] db-new-account-insert : crc-bank proposal modify_date --id 1 --start 07/25/2023 --end 07/26/2023 test

(bank_deploy) [nlc60@moss bank] db-new-account-insert : crc-bank account info test
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                         Proposal End Date:                        |          07/26/2024          |                             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                            Proposal ID:                           |              2               |                             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                                                                   |                              |                             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                            Cluster: GPU                           |     Available SUs: 10000     |                             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                                User                               |           SUs Used           |     Percentage of Total     |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                               nlc60                               |              0               |             N/A             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                          Overall for GPU                          |              0               |              0              |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                                                                   |                              |                             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                            Cluster: SMP                           |     Available SUs: 10000     |                             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                                User                               |           SUs Used           |     Percentage of Total     |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                               nlc60                               |              0               |             N/A             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                          Overall for SMP                          |              0               |              0              |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                                                                   |                              |                             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                       Floating Service Units                      |       Units Remaining        |         Percent Used        |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|     **Floating SUs are applied to any cluster to cover usage      |              0               |              0              |
|                     exceeding proposal limits                     |                              |                             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
|                          Aggregate Usage                          |              0               |                             |
+-------------------------------------------------------------------+------------------------------+-----------------------------+
```

Technically, reducing the available SUs for the allocations would accomplish this as well, but either method involves manually changing database entries from their original values. 

In that our start date's resolution is days, we may want to sort by descending proposal ID when looking for the active proposal throughout the application.

**Comment by Comeani** [comment]
I found that there are in fact tests for checking overlapping proposals, they're just passing because the proposal start dates are not registering as equal:
```
(Pdb) query = select(Proposal).where(Proposal.account_id.in_(overlapping_proposal_sub_1))
(Pdb) session.execute(query).scalars().all()
[<bank.orm.Proposal object at 0x7fc8cf568520>, <bank.orm.Proposal object at 0x7fc8cf568df0>, <bank.orm.Proposal object at 0x7fc8cf568670>, <bank.orm.Proposal object at 0x7fc8cf568040>, <bank.orm.Proposal object at 0x7fc8cf568070>, <bank.orm.Proposal object at 0x7fc8cf5684c0>]
(Pdb) props = session.execute(query).scalars().all()
(Pdb) props[1]
<bank.orm.Proposal object at 0x7fc8cf568bb0>
(Pdb) props[1].start_date
datetime.date(2023, 7, 26)
(Pdb) props[1].start_date == start
False
(Pdb) start
datetime.datetime(2023, 8, 1, 17, 22, 11, 446469)
(Pdb) date.today()
datetime.date(2023, 8, 1)
```

**Comment by Comeani** [comment]
Fixed by #315
